sound/soc/sof/topology.c

```diff
@@ -492,6 +492,16 @@ static int get_token_u16(void *elem, void *object, u32 offset, u32 size)
 	return 0;
 }
 
+static int get_token_uuid(void *elem, void *object, u32 offset, u32 size)
+{
+	struct snd_soc_tplg_vendor_uuid_elem *velem = elem;
+	guid_t *dst = (guid_t *)((u8 *)object + offset);
+
+	guid_copy(dst, (guid_t *)velem->uuid);
```

Comment:
@keyonjie Is this correct, I thought we are not using the kernrel guid_t? This seems a bit dangerous as the definition is uint8_t[16] and here we treat it as guid_t.

Reply:
> @keyonjie Is this correct, I thought we are not using the kernrel guid_t? This seems a bit dangerous as the definition is uint8_t[16] and here we treat it as guid_t.

@kv2019i  the guid_t is actually 16 unsigned bytes, so your point here is?
```
typedef struct {
	__u8 b[16];
} guid_t;
```

Reply:
@keyonjie  That's the definition now, but if at some point guid_t definition is changed, then SOF is broken. So either we should use guid_t in also the IPC definition, or not use it. guid_t is perhaps more stable than most, so in practise this is probably a minor problem, but I don't really see why we need guid_t at all if you are just passing bytes around.

Reply:
please don't ignore previous feedback.

https://github.com/thesofproject/linux/pull/2090#discussion_r423313028

Reply:
> please don't ignore previous feedback.
> 
> [#2090 (comment)](https://github.com/thesofproject/linux/pull/2090#discussion_r423313028)

Thanks for reminding, I thought I had changed that but definitely not, sorry about that. As @kv2019i mentioned, let me consider changing it to memcpy for bytes.

@lgirdwood are you OK with this, or do you still prefer translating it to struct sof_uuid before transmitting to FW?
```
struct sof_uuid {
	uint32_t a;
	uint16_t b;
	uint16_t c;
	uint8_t  d[8];
};
```